pkg/scorecard_result.go

```diff
@@ -135,9 +155,105 @@ func (r *ScorecardResult) AsString(showDetails bool, writer io.Writer) error {
 	table.SetCenterSeparator("|")
 	table.AppendBulk(data)
 	table.Render()
+
+	return nil
+}
+
+// UPGRADEv2: new code.
+func (r *ScorecardResult) AsString2(showDetails bool, logLevel zapcore.Level, writer io.Writer) error {
```

Comment:
For new code, we can do away with `showDetails` bool. We can use `logLevel` to decide what details to show if any. Especially if we can update `logLevel` to be `DetailType` enum.

Reply:
let's do that in a follow-up CL. This requires some refactoring to handle the `--show-details` argument. Created https://github.com/ossf/scorecard/issues/717 for tracking the issue.